rdsbroker/parameter_groups_source.go

```diff
@@ -0,0 +1,166 @@
+package rdsbroker
+
+import (
+	"code.cloudfoundry.org/lager"
+	"fmt"
+	"github.com/alphagov/paas-rds-broker/awsrds"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"sort"
+	"strings"
+)
+
+//go:generate counterfeiter -o fakes/fake_parameter_group_selector.go . ParameterGroupSelector
+type ParameterGroupSelector interface {
+	SelectParameterGroup(servicePlan ServicePlan, parameters ProvisionParameters) (string, error)
+}
+
+type ParameterGroupSource struct {
+	config                     Config
+	rdsInstance                awsrds.RDSInstance
+	logger                     lager.Logger
+	supportedPreloadExtensions map[string][]DBExtension
+}
+
+func NewParameterGroupSource(config Config, rdsInstance awsrds.RDSInstance, supportedPreloadExtensions map[string][]DBExtension, logger lager.Logger) *ParameterGroupSource {
+	return &ParameterGroupSource{config, rdsInstance, logger, supportedPreloadExtensions}
+}
+
+func (pgs *ParameterGroupSource) SelectParameterGroup(servicePlan ServicePlan, parameters ProvisionParameters) (string, error) {
+	pgs.logger.Debug("selecting a parameter group", lager.Data{
+		servicePlanLogKey: servicePlan,
+		extensionsLogKey:  parameters.Extensions,
+	})
+
+	groupName := composeGroupName(pgs.config, servicePlan, parameters, map[string][]DBExtension{})
```

Comment:
I suspect this should have `pgs. supportedPreloadExtensions` as the last parameter